lnwallet/channel.go

```diff
@@ -2340,18 +2340,20 @@ func createHtlcRetribution(chanState *channeldb.OpenChannel,
 	// We'll generate the original second level witness script now, as
 	// we'll need it if we're revoking an HTLC output on the remote
 	// commitment transaction, and *they* go to the second level.
+	//nolint:ll
 	secondLevelAuxLeaf := fn.FlatMapOption(
 		func(l CommitAuxLeaves) fn.Option[input.AuxTapLeaf] {
-			return fn.MapOption(func(val uint16) input.AuxTapLeaf {
-				idx := input.HtlcIndex(val)
+			return fn.MapOption(
+				func(val tlv.BigSizeT[uint64]) input.AuxTapLeaf {
+					idx := val.Int()
 
-				if htlc.Incoming.Val {
-					leaves := l.IncomingHtlcLeaves[idx]
-					return leaves.SecondLevelLeaf
-				}
+					if htlc.Incoming.Val {
+						leaves := l.IncomingHtlcLeaves[idx]
+						return leaves.SecondLevelLeaf
+					}
 
-				return l.OutgoingHtlcLeaves[idx].SecondLevelLeaf
-			})(htlc.HtlcIndex.ValOpt())
+					return l.OutgoingHtlcLeaves[idx].SecondLevelLeaf
+				})(htlc.HtlcIndex.ValOpt())
```

Comment:
nano nit: I think a line break after the curly brace would be correct here, but feel free to ignore.